File: logistics/src/main/java/com/example/logistics/model/Package.java
package com.example.logistics.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "packages")
@Data
@NoArgsConstructor
public class Package {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id", nullable = false)
    private User seller;

    @Column(nullable = false)
    private Double weightKg;
    @Column(nullable = false)
    private Double sizeWidthCm;
    @Column(nullable = false)
    private Double sizeHeightCm;
    @Column(nullable = false)
    private Double sizeDepthCm;

    @Column(nullable = false)
    private String clientName;
    @Column(nullable = false)
    private String clientAddress;
    @Column(nullable = false)
    private String clientPhone;

    @Column(nullable = false)
    private LocalDate pickupDate;
    
    @Column(nullable = false)
    private Double price;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PackageStatus status;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

enum PackageStatus {
    PENDING, IN_TRANSIT, DELIVERED, CANCELED
}

File: bs/lib/screens/models/package.dart
class Package {
  final String? id;
  final double weightKg;
  final double sizeWidthCm;
  final double sizeHeightCm;
  final double sizeDepthCm;
  final String clientName;
  final String clientAddress;
  final String clientPhone;
  final DateTime pickupDate;
  final double? price;
  final String? status;

  Package({
    this.id,
    required this.weightKg,
    required this.sizeWidthCm,
    required this.sizeHeightCm,
    required this.sizeDepthCm,
    required this.clientName,
    required this.clientAddress,
    required this.clientPhone,
    required this.pickupDate,
    this.price,
    this.status,
  });

  Map<String, dynamic> toJson() {
    return {
      'weightKg': weightKg,
      'sizeWidthCm': sizeWidthCm,
      'sizeHeightCm': sizeHeightCm,
      'sizeDepthCm': sizeDepthCm,
      'clientName': clientName,
      'clientAddress': clientAddress,
      'clientPhone': clientPhone,
      'pickupDate': pickupDate.toIso8601String(),
    };
  }
}
